Route Names and Vehicle Types

its-all-waves opened the pull request:
Address issue #26.

Icons designed by @HauntedHamburger. (Currently using mock icons.)

## Changes
- new icons for ferries, streetcars, buses + rotating arrows to indicate direction of travel
- show vehicle types and route names and colors in the dropdown menu
- show vehicle types on the map

naf419 commented on app/main.jsx:
```diff
@@ -25,49 +38,74 @@ const ROUTES = NortaGeoJson
-    shadowAnchor: null,
-    iconSize: new L.Point(24, 24),
+// These routes don't exist at NORTA.com
+// When a vehicle enters its garage, its route becomes 'U'
```
unfortunately the U route designation also applies to selection of 24 hours routes that are *not* in the garage and running their normal route between 12:30am-ish and 4:00am-sh, so filtering these out isn't currently desirable

its-all-waves replied:
Understood. Thanks for knowing that and for trying to fix it at the source. 

What icon and label should we assign to the U route then?

(The ferry icon is temp, and the dropdown design has not been solidified.)
<img width="359" alt="Screenshot 2024-12-31 at 3 44 45 PM" src="https://github.com/user-attachments/assets/621d1ce8-8612-41fc-a3b8-d48054c15048" />

naf419 commented on app/main.jsx:
```diff
@@ -193,15 +240,21 @@ class App extends React.Component {
-        const routeOptions = routes.map(r => {
-            return { value: r, label: r }
+        const routeOptions = routes.map(rt => {
+            let { name, type } = routesInfo[rt]
```
need `const type = routesInfo[v.rt]?.type ?? 'error'`-ish protection here, as above, to avoid js errors when a downloaded route# doesnt match anything in route_info.json

opportunity to commonize that safe-access default-to-error code, perhaps?

naf419 commented on data/routes_info.json:
script to generate from GTFS.zip maybe?

its-all-waves replied:
Good catch!

its-all-waves commented:
@naf419, thanks for the notes! Should be able to get to this in the coming week or so.

its-all-waves commented:
@naf419, I believe I've addressed your comments. At the moment, I'm still filtering out PI and PO routes, and the U route appears in the dropdown again.

<img width="350" alt="Screenshot 2025-01-02 at 9 40 11 AM" src="https://github.com/user-attachments/assets/ac664fd6-abb6-40e9-9836-85e74fc0f2a6" />

Still don't know how we should handle presenting the U route to the user. It seems we can't tell what vehicle type it's going to be, so I don't know which icon to use, and we don't have a route drawn on the map since U route is not present in routes.json. We also need to figure out how to name the U route in the dropdown.

naf419 commented:
> @naf419, I believe I've addressed your comments. At the moment, I'm still filtering out PI and PO routes, and the U route appears in the dropdown again, as shown above, but I'm not sure where the name "r" came from before. Now that line in the dropdown reads "U <icon> U".
> 
> Still don't know how we should handle presenting the U route to the user. It seems we can't tell what vehicle type it's going to be, so I don't know which icon to use, and we don't have a route drawn on the map since U route is not present in routes.json. We also need to figure out how to name the U route in the dropdown.

i only currently log route #s, not types, to support my bug report to rta/clever [lotta good that did me])
but i see no reason to assume without any additional info that the route_type would be incorrect as well, so i would suggest (grain of salt) to just show whatever route_type it says even for the U route, and maybe route "U" gets displayed as route "Unknown"?

any thoughts for/against showing the route# instead of an icon (no shade but my gut tells me that the one on the river is the ferry and the one on the streetcar tracks is the streetcar, i just wanna know which of the 3 routes ive already filtered down to is which without clicking it when they all start at the library or whatever)

its-all-waves commented:
@naf419, do we know if there is one ferry per ferry route? I am getting 2 ferries from the true Bustime api -- 1 for Algiers Point and 1 for Chalmette (of course this is during normal hours, not overnight). (I don't see any data related to route type in the Bustime responses -- seems route type is coming from routes.json. Assuming you're logging Bustime responses.)

<img width="601" alt="Screenshot 2025-01-02 at 12 44 29 PM" src="https://github.com/user-attachments/assets/05b4baaa-dccf-4fa8-a685-6d142a8336fb" />

> any thoughts for/against showing the route# instead of an icon? no shade but my gut tells me that the one on the river is the ferry and the one on the streetcar tracks is the streetcar,

Do you mean removing icon for non-U routes as well? Or remove the icon specifically for U routes? I think I'm misunderstanding because we would need some icon on the map anyway... So it would make sense to use the same icon in the dropdown. Unless we revert to using a plain arrow again.

To be clear, I will continue ignoring PI and PO routes until someone says otherwise.

naf419 commented:
> @naf419, do we know if there is one ferry per ferry route? I am getting 2 ferries from the true Bustime api -- 1 for Algiers Point and 1 for Chalmette (of course this is during normal hours, not overnight).

> Edit: Looks like there is one ferry per route according to the [ferry schedule pdf](https://www.norta.com/RTA/media/RTABusPDFSchedules/route_1.pdf).

generally, yes that is my understanding.  [if we wanted to get into the weeds, there is another ferry that sits docked in algiers, but i think nowadays its only ever used for special events (i.e. to go from canal to gretna for gretna fest), and i have doubts that this special event route is reflected in data at all even during these special events] 

> (I don't see any data related to route type in the Bustime responses -- seems route type is coming from routes.json. Assuming you're logging Bustime responses.)

yea sorry i had the data sources mixed up when i incorrectly brushed off your question of 'what icon to show for U routes'.
your question makes a lot more sense when, as you say, we dont know the type because we dont know the route. got it.

in that case ill vote for a question mark for unknown route types i guess? (i think it would still be clear no matter what any chosen error/unknown icons in the streetcar barn are streetcars in for the night and error/unknown icon driving a regular bus route are the rogue broken-route#-latenight buses)?

> > any thoughts for/against showing the route# instead of an icon? no shade but my gut tells me that the one on the river is the ferry and the one on the streetcar tracks is the streetcar,
> 
> Do you mean removing icon for non-U routes as well? Or remove the icon specifically for U routes? I think I'm misunderstanding because we would need some icon on the map anyway... So it would make sense to use the same icon in the dropdown. Unless we revert to using a plain arrow again, in which case we don't need icons in the dropdown at all.

yes, the questions is: instead of having (current) just-an-arrow icon for all vehicle types, or (your implementation here) vehicle-type-specific icons, would it be better to have route#-icons. like a little "57" in the dot/bubble on the map [without clicking a vehicle to see its route#], so i can see its the 57 bus and not the 11 bus when i have a filter that shows both. not as pretty as your vehicle-type-specific approach of course, but IMO more useful. and technically this would solve the "what icon to show for U routes" problem as well, just show a U :-) just a thought.

its-all-waves commented:
@naf419, 

> would it be better to have route#-icons with direction arrow. like a little "57" in the dot/bubble on the map [without clicking a vehicle to see its route#], so i can see its the 57 bus and not the 8 bus when i have a filter that shows both

I see where you're coming from and I like that it solves the U route icon problem. For the map icons, maybe there's a subtle way to communicate the vehicle type while prioritizing the route number. I'll be thinking about that, but in the meantime I'll ditch the vehicle-specific icons and go with route numbers on the map, and keep vehicle type icons in the dropdown menu, as you suggested.

bhelx commented:
>  generally, yes that is my understanding. [if we wanted to get into the weeds, there is another ferry that sits docked in algiers, but i think nowadays its only ever used for special events (i.e. to go from canal to gretna for gretna fest), and i have doubts that this special event route is reflected in data at all even during these special events]

95% of the time there is one ferry in rotation. during carinval time, and special events like jazz fest, nye, etc, they run two at the same frequency, but staggered 15 mins. I don't think they ever reflect this in the GTFS data, but i'm not sure.

naf419 commented:
> I see where you're coming from and I like that it solves the U route icon problem. For the map icons, maybe there's a subtle way to communicate the vehicle type while prioritizing the route number. I'll be thinking about that, but in the meantime I'll ditch the vehicle-specific icons and go with route numbers on the map, and keep vehicle type icons in the dropdown menu, as you suggested.

@its-all-waves , while we think on the icons on the map piece of this, I went ahead and resolved all the merge conflicts with this draft PR and included it in a new PR (with the new map icon code present but unused) just to get the route names and icons in the dropdown menu

naf419 commented:
Here's my latest attempt at conveying both route# and direction on map:
![image](https://github.com/user-attachments/assets/9e13f2ed-5dad-4b88-b26f-88047f5ccf8c)

yes it looks like shit because im both shit at aesthetics and at css
